src/ucp/wireup/wireup_cm.c

```diff
@@ -1197,27 +1215,31 @@ static ssize_t ucp_cm_server_priv_pack_cb(void *arg,
     }
 
     if (worker->cms[ucp_ep_ext_control(ep)->cm_idx].attr.max_conn_priv <
-        (sizeof(*sa_data) + ucp_addr_size)) {
+        ucp_cm_priv_data_length(ucp_addr_size)) {
         status = UCS_ERR_BUFFER_TOO_SMALL;
         goto free_addr;
     }
 
+    data_buf_size = ucp_cm_priv_data_length(ucp_addr_size);
+    data_buf      = ucs_malloc(data_buf_size, "server_priv_data");
+    if (data_buf == NULL) {
+        status = UCS_ERR_NO_MEMORY;
+        goto free_addr;
+    }
+
     dev_index = ucp_cm_tl_bitmap_get_dev_idx(worker->context, tl_bitmap);
-    ucp_cm_priv_data_pack(sa_data, ep, dev_index, ucp_addr, ucp_addr_size);
+    ucp_cm_priv_data_pack(data_buf, ep, dev_index, ucp_addr, ucp_addr_size);
+
+    *data_buf_p      = data_buf;
+    *data_buf_size_p = data_buf_size;
+    status           = UCS_OK;
```

Comment:
can we unify common code with ucp_cm_client_resolve_cb?

Reply:
yes, it could be nice refactoring but much better to do this when tmp ep is removed because on client side we use real EP to pack UCP address but tmp EP to pack sa_data. This make the code ugly for now.

Reply:
ok